### PR TITLE
SAAS-3752: Added Oracle Admin User API Keys plugin, spec

### PR DESCRIPTION
**Pull request by AkhtarAmir**

### CODE REVIEW

File: plugins/oracle/identity/adminUserAPIKeys.js [hunk 1]
```diff
@@ -0,0 +1,105 @@
...
+
+module.exports = {
+    title: 'Admin User API Keys',
+    category: 'Identity',
+    domain: 'Identity and Access Management',
+    description: 'Ensure that API keys do not exist for tenancy administrator users.',
+    more_info: 'The administrator user should avoid using API keys. Since the administrator user has full permissions across the entire tenancy, creating API keys for it only increases the chance that they are compromised. ' +
+        'Instead, create non-admin user with limited permissions and use its API keys.',
+    link: 'https://docs.oracle.com/en/cloud/get-started/subscriptions-cloud/csgsg/create-users-and-assign-roles.html',
+    recommended_action: 'Enable an MFA device for the user account.',
```
Comment (AkhtarAmir) [inline plugins/oracle/identity/adminUserAPIKeys.js:11]:
```suggestion
    recommended_action: 'Remove API keys for administrator users',
```

File: plugins/oracle/identity/adminUserAPIKeys.js [hunk 2]
```diff
@@ -0,0 +1,105 @@
...
+        if (!apiKeys.data.length) {
+            helpers.addResult(results, 0, 'No user API keys found', 'global');
+            return callback(null, results, source);
+        }
+
+        let adminGroup = groups.data.find(group => group.name == 'Administrators');
+        let apiKeyIds = apiKeys.data.map(apiKey => apiKey.userId) || [];
+
+        if (adminGroup) {
+            adminGroupId = adminGroup.id;
```
Comment (dipsubha06) [inline plugins/oracle/identity/adminUserAPIKeys.js:82]: we are not using this any where?

File: plugins/oracle/identity/adminUserAPIKeys.js [hunk 3]
```diff
@@ -0,0 +1,105 @@
...
+
+            let adminUsers = userGroups.data.map(userGroup => userGroup.userId) || [];
```
Comment (dipsubha06) [inline plugins/oracle/identity/adminUserAPIKeys.js:84]: This api data will always give userid of admin users? no need to check for any flag or something?

File: plugins/oracle/identity/adminUserAPIKeys.js (hunk 2 above)
Comment (AkhtarAmir) [inline plugins/oracle/identity/adminUserAPIKeys.js:82]:
```suggestion
```

File: plugins/oracle/identity/adminUserAPIKeys.js (hunk 3 above)
Comment (AkhtarAmir) [reply plugins/oracle/identity/adminUserAPIKeys.js:83]: If it doesn't, userGroup.userId will be undefined but code won't break

File: plugins/oracle/identity/usersMfaEnabled.js [hunk 4]
```diff
@@ -8,7 +8,7 @@ module.exports = {
     more_info: 'User accounts should have an MFA device setup to enable two-factor authentication.',
     link: 'https://docs.cloud.oracle.com/iaas/Content/Identity/Tasks/usingmfa.htm',
     recommended_action: 'Enable an MFA device for the user account.',
-    apis: ['user:list'],
+    apis: ['user:list', 'apiKey:list'],
```
Comment (AkhtarAmir) [inline plugins/oracle/identity/usersMfaEnabled.js:11]:
```suggestion
    apis: ['user:list'],
```

File: plugins/oracle/identity/usersMfaEnabled.js [hunk 5]
```diff
@@ -33,6 +33,7 @@ module.exports = {
             warn_federated_users: settings.warn_federated_users || this.setting.warn_federated_users.default
         };
         var warnFedUsers = (config.warn_federated_users == 'true');
+        console.log(JSON.stringify(cache, null, 2));
```
Comment (AkhtarAmir) [inline plugins/oracle/identity/usersMfaEnabled.js:36]:
```suggestion
```

**Pull request merged**